### DISCUSSION TITLE
cannot pickle '_thread.RLock' object

### DISCUSSION THREAD

**Discussion by aradhanachaturvedi** [mlflow/mlflow#12347] [Q&A]
I am using mlflow.pyfunc.log_model however getting the response as cannot pickle '_thread.RLock' object.What can I do to avoid this ?is it happenign because of chromadb/OpenAI??

class MyVanna(ChromaDB_VectorStore, OpenAI_Chat):
    def __init__(self, config=None):
        ChromaDB_VectorStore.__init__(self, config=config)
        OpenAI_Chat.__init__(self, client=llm, config=config) # Make sure to put your AzureOpenAI client here
    def run_sql_function(self, question: str):
            sql = self.generate_sql(question=question)
            df = self.run_sql(sql=sql)
            questions = self.generate_questions()
            natural_language_summary = self.generate_summary(question=question, df=df)
            followup_questions = self.generate_followup_questions(question=question, sql=sql, df=df)
            if followup_questions is not None and len(followup_questions) > 5:
                followup_questions = followup_questions[:5]
            return {"answer": natural_language_summary, "followup": followup_questions}

class MyVannaModel(mlflow.pyfunc.PythonModel):
    def load_context(self, context):
        pass
    
    def predict(self, context, model_input):
        # model_input is expected to be a pandas DataFrame with one column 'question'
        
        vanna = MyVanna(config={'model': 'gpt-4o'})
        vanna.connect_to_postgres(
            host=os.getenv("POSTGRES_HOST"),
            dbname=os.getenv("POSTGRES_DATABASE"),
            user=os.getenv("POSTGRES_USER"),
            password=os.getenv("POSTGRES_PASSWORD"),
            port=port
        )
        question =model_input['input']
        response = vanna.run_sql_function(question=question)
        
        response =get_response(model_input)
        
        return response

# Log the model to create a serving endpoint
model = MyVannaModel()

mlflow.pyfunc.log_model( artifact_path = 'model',python_model=MyVannaModel registered_model_name="vanna api")

**Comment by AmrAbdelghanyy** [comment]
Getting same error while logging a custom python rasa model, not sure what is causing the issue.
